### PR TITLE
SKIP-227 - Crash in skippy_m3u8_client_get_target_duration

### PR DESCRIPTION
**Pull request by milos-pesic-sc**
- Covering the case when playlist buffer does not contain expected data.

### CODE REVIEW

File: src/skippy_m3u8.c [hunk 1]
```diff
@@ -755,6 +755,7 @@ skippy_m3u8_client_get_target_duration (SkippyM3U8Client * client)
   GstClockTime duration = 0;
 
   g_return_val_if_fail (client != NULL, GST_CLOCK_TIME_NONE);
+  g_return_val_if_fail (client->current != NULL, GST_CLOCK_TIME_NONE);
```
Comment (milos-pesic-sc) [inline src/skippy_m3u8.c]: We do not necessary need this check now - but I am adding it just to be consistent and on safe side.

File: src/skippy_m3u8.c [hunk 2]
```diff
@@ -1010,6 +1011,7 @@ skippy_m3u8_client_load_playlist (SkippyM3U8Client * client, GstBuffer* playlist
   gchar* playlist = buf_to_utf8_playlist (playlist_buffer);
   if (playlist == NULL) {
     GST_WARNING ("Error converting playlist from raw buffer to UTF8");
+    return FALSE;
```
Comment (milos-pesic-sc) [inline src/skippy_m3u8.c]: This is only thing we need to do in addition. I mentioned posting error message to the bus but this is done by caller of this function so we do not need to worry about that.

**Comment by milos-pesic-sc** [comment]
@pokey909 @tchakabam - please take a look whenever you have time.

File: src/skippy_m3u8.c (hunk 1 above)
Comment (tchakabam) [inline src/skippy_m3u8.c]: :+1:

File: src/skippy_m3u8.c (hunk 2 above)
Comment (pokey909) [inline src/skippy_m3u8.c]: Should GST_WARNING changed to GST_ERROR here? Since we treat it like an error.

**Comment by tchakabam** [comment]
awesooome!!! :+1:

**Pull request merged**